sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -1942,3 +1963,94 @@ def __getitem__(self, snapshot_id: SnapshotId) -> Snapshot:
         if snapshot is None:
             raise KeyError(snapshot_id)
         return snapshot
+
+
+class SharedVersionSnapshot(PydanticModel):
+    """A stripped down version of a snapshot that is used for fetching snapshots that share the same version
+    with a significantly reduced parsing overhead.
+    """
+
+    name: str
+    version: str
+    temp_version_: t.Optional[str] = Field(alias="temp_version")
+    identifier: str
+    fingerprint: SnapshotFingerprint
+    interval_unit: IntervalUnit
+    change_category: SnapshotChangeCategory
+    updated_ts: int
+    unpaused_ts: t.Optional[int]
+    unrestorable: bool
+    disable_restatement: bool
+    effective_from: t.Optional[TimeLike]
+    raw_snapshot: t.Dict[str, t.Any]
+
+    @property
+    def snapshot_id(self) -> SnapshotId:
+        return SnapshotId(name=self.name, identifier=self.identifier)
+
+    @property
+    def is_forward_only(self) -> bool:
+        return self.change_category == SnapshotChangeCategory.FORWARD_ONLY
+
+    @property
+    def normalized_effective_from_ts(self) -> t.Optional[int]:
```

Comment:
should we make a mixin for these shared methods or no, not worth it?

Reply:
It's the only one. Didn't want to bother for now, but I'm open to reconsider